client/src/components/RelatedProducts/ProductCard.jsx

```diff
@@ -4,72 +4,156 @@ import { makeStyles } from '@material-ui/core/styles';
+  },
+  icon: {
+    position: 'relative to parent',
+    top: 5,
   }
 }));
 
 const ProductCard = (props) => {
   const [productCardInfo, setProductCardInfo] = useState(null);
   const [productCardPhoto, setProductCardPhoto] = useState(null);
+  const [ratingsInfo, setRatingsInfo] = useState({
+          avgProductRating: 0,
+          totalRatings: 0
+          });
+  const [currentProductInfo, setCurrentProductInfo] = useState(null);
+  const [showComparison, setShowComparison] = useState(false);
+  const [featureData, setFeatureData] = useState(null);
   const classes = useStyles();
 
+
   const getPhoto = () => {
     axios.get(`/api/products/${props.productId}/styles`)
-      .then(stylesInfo => {
-        setProductCardPhoto(stylesInfo.data.results[0]);
-      })
-      .catch(err => {
-        console.info('There was an error getting product photo from the server.');
-      });
+    .then(stylesInfo => {
+      setProductCardPhoto(stylesInfo.data.results[0]);
+    })
+    .catch(err => {
+      console.info('There was an error getting product photo from the server.');
+    });
   }
 
   const getProductInfo = () => {
     axios.get(`/api/products/${props.productId}`)
-      .then(productInfo => {
-        setProductCardInfo(productInfo.data);
-      })
-      .catch(err => {
-        console.info('There was an error retrieving product information from the server.');
-      });
+    .then(productInfo => {
+      setProductCardInfo(productInfo.data);
+    })
+    .catch(err => {
+      console.info('There was an error retrieving product information from the server.');
+    });
+  }
+
+  const getCurrentProductInfo = () => {
+    axios.get(`/api/products/${props.currentProduct.id}`)
+    .then(productInfo => {
+      setCurrentProductInfo(productInfo.data);
+    })
+    .catch(err => {
+      console.info('There was an error retrieving product information from the server.');
+    });
+  }
+
+  const getFeatureData = () => {
+    const featureData = {};
+
+    for (var i = 0; i < currentProductInfo.features.length; i++) {
+      var currentFeatureObject = currentProductInfo.features[i];
+      var currentFeatureValue = [];
+      if (currentFeatureObject.value === null) {
+        currentFeatureValue.push('yes!');
+      } else {
+        currentFeatureValue.push(currentFeatureObject.value);
+      }
+      currentFeatureValue.push('');
+      featureData[currentFeatureObject.feature] = currentFeatureValue;
+    }
+
+    for (var j = 0; j < productCardInfo.features.length; j++) {
+      var currentFeatureObject = productCardInfo.features[j];
+      if (featureData[currentFeatureObject.feature] !== undefined) {
+        if (currentFeatureObject.value === null) {
+          featureData[currentFeatureObject.feature] = 'yes!';
+        } else {
+          featureData[currentFeatureObject.feature][1] = currentFeatureObject.value;
+        }
+      } else {
+        var currentFeatureValue = [];
+        currentFeatureValue.push('');
```

Comment:
Is there a reason why you're pushing an empty string?

Reply:
Yes! When I am populating my comparison modal, I need a placeholder for that column.